src/python/gem5/simulate/simulator.py

```diff
@@ -185,6 +190,38 @@ def unique_exit_event():
 
         self._checkpoint_path = checkpoint_path
 
+    def schedule_simpoint(
+        self,
+        simpoint_start_insts: List[int] = [],
+        schedule_at_init: bool = False
+        )-> None:
+        """
+        Schedule SIMPOINT_BEGIN exit events
+
+        :param simpoint_start_insts: a list of number of instructions
+        indicating the starting point of the simpoints
+        :param schedule_at_init: if it is True, schedule the events in the init
+        stage of the core, else, schedule the events during the simulation
+        """
+        self._board.get_processor().get_cores()[0].\
+                        set_simpoint(simpoint_start_insts,schedule_at_init)
+
+    def schedule_max_insts(
```

Comment:
For this function, we definitely don't want to assume just 1 core. I'm not certain I know *what* we want to do, though.

I think maybe this should be a board function / processor function.

Reply:
Maybe it should be a processor function because it's more related to a processor than to a board..?

Reply:
I'll drop this comment. I'm trying to look for too much "perfection" and I'm not really sure there's a better way.